src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -1490,6 +1619,8 @@ static int parse_args(std::vector<const char*>& args,
       cfg->pretty_format = true;
     } else if (ceph_argparse_flag(args, i, "--remap-failure-fatal", (char *)NULL)) {
       cfg->remap_failure_fatal = true;
+    } else if (ceph_argparse_flag(args, i, "--adapter-monitoring-enabled", (char *)NULL)) {
+      cfg->remap_failure_fatal = true;
```

Comment:
Makes me suspicious of the final build's testing ;)
```suggestion
      cfg->adapter_monitoring_enabled = true;
```

Reply:
Really sorry about this, looks like I'm hitting some weird Windows behavior. I'm replacing the executable, the file hash is the right one, however when executed it reports an older version:

```
PS Microsoft.PowerShell.Core\FileSystem::\\13.13.13.13\workspace\ceph.windows\build\bin> .\rbd-wnbd --version
ceph version 18.0.0-3084-gf73948acb72 (f73948acb72ccecfe085ce1122219ee29a9109fb) reef (dev)

PS Microsoft.PowerShell.Core\FileSystem::\\13.13.13.13\workspace\ceph.windows\build\bin> get-filehash .\rbd-wnbd.exe

Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          18CAAC0CBB0D45E7F08AB12E2E2A9A60792C365EE0417F850D8B9014931131C0       \\13.13.13.1...


PS Microsoft.PowerShell.Core\FileSystem::\\13.13.13.13\workspace\ceph.windows\build\bin> cp .\rbd-wnbd.exe 'C:\Program Files\Ceph\bin'
PS Microsoft.PowerShell.Core\FileSystem::\\13.13.13.13\workspace\ceph.windows\build\bin> get-filehash 'C:\Program Files\Ceph\bin\rbd-wnbd.exe'

Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          18CAAC0CBB0D45E7F08AB12E2E2A9A60792C365EE0417F850D8B9014931131C0       C:\Program F...


PS Microsoft.PowerShell.Core\FileSystem::\\13.13.13.13\workspace\ceph.windows\build\bin> & 'C:\Program Files\Ceph\bin\rbd-wnbd.exe' --version
ceph version 18.0.0-3083-gcced38b6116 (cced38b6116eb8d8a380ae1f59ea1f3aee168960) reef (dev)

# same file hash, different version reported when executed
```

It's probably using a binary cached in memory for some reason, despite not having any running ``rbd-wnbd`` processes.

Reply:
Double tested the latest version:

* without monitoring

```
2023-03-24T08:54:00.627Coordinated Universal Time 3  0 rbd-wnbd: WNBD adapter monitoring disabled.
2023-03-24T08:54:00.627Coordinated Universal Time 3  0 Successfully started service.
```

* with monitoring

```
2023-03-24T08:55:43.215Coordinated Universal Time c  5 rbd-wnbd: monitor_wnbd_adapter: initializing COM
2023-03-24T08:55:43.215Coordinated Universal Time 3  0 Successfully started service.
2023-03-24T08:55:43.215Coordinated Universal Time d  5 rbd-wnbd: Accepting admin pipe connections.
2023-03-24T08:55:43.215Coordinated Universal Time c  5 rbd-wnbd: monitor_wnbd_adapter: initializing wmi subscription
2023-03-24T08:55:44.114Coordinated Universal Time c  0 rbd-wnbd: monitoring wnbd adapter state changes
2023-03-24T08:55:56.148Coordinated Universal Time c  0 rbd-wnbd: wnbd adapter removed
2023-03-24T08:55:58.523Coordinated Universal Time c  0 rbd-wnbd: wnbd adapter (re)created, remounting disks
2023-03-24T08:55:58.539Coordinated Universal Time c  0 rbd-wnbd: remounting persistent disks
2023-03-24T08:55:58.539Coordinated Universal Time e  1 rbd-wnbd: Remapping: rbd/test_4g_1. Timeout: 20000 ms.
2023-03-24T08:55:58.539Coordinated Universal Time f  1 rbd-wnbd: Remapping: rbd/test_4g_2. Timeout: 20000 ms.
2023-03-24T08:55:58.539Coordinated Universal Time 10  1 rbd-wnbd: Remapping: rbd/test_4g_3. Timeout: 20000 ms.
2023-03-24T08:55:58.539Coordinated Universal Time 11  1 rbd-wnbd: Remapping: rbd/test_4g_4. Timeout: 20000 ms.
```